WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/ServiceRequestContext.cs

```diff
@@ -179,11 +187,15 @@ public bool IsNewSessionId(string sessionId)
                                                    fileName),
                                       requestTaskOptions.ApplicationNamespace);
 
-    if (ServicesMapper.ContainsKey(serviceId.Key))
+    if (currentService_?.ServiceId == serviceId)
     {
-      return serviceId;
+      return currentService_;
     }
 
+    logger_.LogInformation($"Worker needs to load new context, from {currentService_?.ServiceId} to {serviceId}");
+
+    currentService_?.Dispose();
```

Comment:
You should set it back to null after the dispose, in case there is an exception before it is reset.